app/main.js

```diff
@@ -35,8 +41,38 @@ function createWindow () {
   mainWindow.on("closed", function () {
     mainWindow = null
   })
+
+  ipcMain.on("initialized", () => {
+    if (deepLinkURLOnReady != null) {
+      // If open-url event was fired before app was ready
+      loadPopulatePage(deepLinkURLOnReady)
+    }
+  })
+
+  ipcMain.on("requestAuthorization", (e, assignmentURL) => {
+    authorizeUser(mainWindow, assignmentURL)
+  })
+}
+
+function loadPopulatePage (assignmentURL) {
+  mainWindow.webContents.send("open-url", assignmentURL)
 }
 
+app.on("open-url", function (event, urlToOpen) {
+  event.preventDefault()
+  var urlParams = new URL(urlToOpen).searchParams
+  if (urlParams.has("assignment_url")) { // Classroom sent deep link
```

Comment:
:art: feel free to introduce some local variables to avoid the need to add lots of comments to explain things:

```js
var urlParams = new URL(urlToOpen).searchParams
var hasDeepLink = urlParams.has("assignment_url")
if (hasDeepLink) { 
 // ...
}
```